trunk/public/class-facebook-login-public.php

```diff
@@ -556,12 +599,14 @@ private function generateUsername( $user ) {
 
 		// User name can't be on the blacklist or empty
 		$illegal_names = get_site_option( 'illegal_names' );
-		if ( empty( $username ) || in_array( $username, (array) $illegal_names ) ) {
+		if ( empty( $username ) || in_array( $username, (array) $illegal_names, true ) ) {
 			// we used all our options to generate a nice username. Use id instead
 			$username = 'fbl_' . $user['id'];
 		}
 
-		// "generate" unique suffix
+		// "Generate" unique suffix. Finds out how many existing users are in the database with the given username (minus numerical suffix), and appends that number to the new username as a pseudounique suffix.
+        // Eg. "SELECT 1 + SUBSTR(user_login, 9) FROM wp_users WHERE user_login REGEXP '^shannon(-[0-9]+)?$' ORDER BY 1 DESC LIMIT 1".
+        // If user 'shannon' exists, result is one field in one row containing "1". That number is appended to username.
 		$suffix = $wpdb->get_var( $wpdb->prepare(
```

Comment:
This is actually not that important. we can just have any unique id, what really matters is the first and last name. The user will never login using his email or username, they'll always login through facebook.

Reply:
Yes, but if we are asking the owner of this project to merge in our changes, they may not agree. Ravi had a suggestion here for changing the logic  - https://10up.slack.com/archives/C105936H3/p1528779474000233 - it's not simpler but it doesn't do that heavy database call. What you think?

Reply:
Right, let's add Ravi's solution for now.

Reply:
Done. 👍